File: src/main/java/com/severett/chargerapp/data/ChargingSessionRepo.java
package com.severett.chargerapp.data;

import com.severett.chargerapp.model.ChargingSession;

import java.util.Optional;
import java.util.UUID;

public interface ChargingSessionRepo {

    ChargingSession save(ChargingSession chargingSession);

    Optional<ChargingSession> findById(UUID id);

    Iterable<ChargingSession> findAll();

}

File: src/main/kotlin/com/severett/chargerapp/data/ChargingSessionRepo.kt
package com.severett.chargerapp.data

import com.severett.chargerapp.model.ChargingSession
import java.util.*

interface ChargingSessionRepo {

    fun save(chargingSession: ChargingSession): ChargingSession

    fun findById(id: UUID): Optional<ChargingSession>

    fun findAll(): Iterable<ChargingSession>

}
